Remotion/Data/DomainObjects.UnitTests/Tracing/TracingDbTransactionTest.cs

```diff
@@ -174,5 +180,31 @@ public void GetIsolationLevel ()
       _innerTransactionMock.Verify();
       _extensionMock.Verify();
     }
+
+    [Test]
+    public void NoFinalizerImplemented ()
+    {
+      var type = typeof(TracingDbTransaction);
+      var finalizer = type.GetMethod("Finalize", BindingFlags.Instance | BindingFlags.NonPublic);
+      Assert.That(finalizer?.DeclaringType, Is.EqualTo(typeof(object)));
+    }
+
+    [Test]
+    public void TestAllVirtualMethodsOverridden ()
+    {
+      var derivedType = typeof(TracingDbTransaction);
+      var baseType = derivedType.BaseType!;
+
+      var virtualMethods = baseType.GetMethods(BindingFlags.Public | BindingFlags.Instance | BindingFlags.DeclaredOnly)
+          .Where(m => m.IsVirtual && !m.IsFinal).ToList();
+
+      foreach (var virtualMethod in virtualMethods)
+      {
+        var parameterTypes = virtualMethod.GetParameters().Select(p => p.ParameterType).ToArray();
+        var overriddenMethod = derivedType.GetMethod(virtualMethod.Name, BindingFlags.Public | BindingFlags.Instance, null, parameterTypes, null);
+        Assert.That(overriddenMethod, Is.Not.Null, $"Method {virtualMethod.Name} not overridden in {derivedType.Name}");
+        Assert.That(overriddenMethod.DeclaringType == derivedType, $"Method {virtualMethod.ToString()} not declared in {derivedType.Name}");
```

Comment:
```suggestion
        Assert.That(overriddenMethod.DeclaringType, Is.EqualTo(derivedType), $"Method {virtualMethod.ToString()} not declared in {derivedType.Name}");
```
Also in the other test fixtures